core/server/policy_violations.go

```diff
@@ -0,0 +1,263 @@
+		FieldSelector: fieldSelector,
+	})
+	opts = append(opts, sigsClient.InNamespace(v1.NamespaceAll))
+
+	validationsList, err := cs.listEvents(ctx, clustersClient, m.ClusterName, false, opts)
+	if err != nil {
+		return nil, fmt.Errorf("error getting events: %v", err)
+	}
+	respErrors = append(respErrors, validationsList.Errors...)
+	policyviolationlist := pb.ListPolicyValidationsResponse{
+		Total:         int32(len(validationsList.Validations)),
+		Violations:    validationsList.Validations,
+		Errors:        respErrors,
+		NextPageToken: validationsList.Token,
+	}
+	return &policyviolationlist, nil
+}
+
+func (cs *coreServer) GetPolicyValidation(ctx context.Context, m *pb.GetPolicyValidationRequest) (*pb.GetPolicyValidationResponse, error) {
+	clusterClient, err := cs.clustersManager.GetImpersonatedClient(ctx, auth.Principal(ctx))
+	if err != nil {
+		return nil, fmt.Errorf("error getting impersonating client: %w", err)
+	}
+
+	selector, err := k8sLabels.ValidatedSelectorFromSet(map[string]string{
+		"pac.weave.works/type": "Admission",
+		"pac.weave.works/id":   m.ViolationId})
+
+	if err != nil {
+		return nil, fmt.Errorf("error building selector for events query: %v", err)
+	}
+	opts := []sigsClient.ListOption{}
+
+	fields := k8sFields.OneTermEqualSelector("type", "Warning")
+	opts = append(opts, &sigsClient.ListOptions{
+		LabelSelector: selector,
+		FieldSelector: fields,
+	})
+	opts = append(opts, sigsClient.InNamespace(v1.NamespaceAll))
+
+	validationsList, err := cs.listEvents(ctx, clusterClient, m.ClusterName, true, opts)
+	if err != nil {
+		return nil, fmt.Errorf("error getting events: %v", err)
+	}
+	if len(validationsList.Errors) > 0 {
+		return nil, fmt.Errorf("error getting events: %s", validationsList.Errors[0].Message)
+	}
+	if len(validationsList.Validations) == 0 {
+		return nil, fmt.Errorf("no policy violation found with id %s and cluster: %s", m.ViolationId, m.ClusterName)
+	}
+	return &pb.GetPolicyValidationResponse{
+		Violation: validationsList.Validations[0],
+	}, nil
+}
+
+func (cs *coreServer) listEvents(ctx context.Context, clusterClient clustersmngr.Client, clusterName string, extraDetails bool, opts []sigsClient.ListOption) (*validationList, error) {
+	respErrors := []*pb.ListError{}
+	clist := clustersmngr.NewClusteredList(func() sigsClient.ObjectList {
+		return &v1.EventList{}
+	})
+
+	if err := clusterClient.ClusteredList(ctx, clist, true, opts...); err != nil {
+		var errs clustersmngr.ClusteredListError
+		if !errors.As(err, &errs) {
+			return nil, fmt.Errorf("error while listing events: %w", err)
+		}
+
+		for _, e := range errs.Errors {
+			respErrors = append(respErrors, &pb.ListError{ClusterName: e.Cluster, Message: e.Err.Error()})
+		}
+	}
+
+	var validations []*pb.PolicyValidation
+	for listClusterName, lists := range clist.Lists() {
+		if clusterName != "" && listClusterName != clusterName {
+			continue
+		}
+		for _, l := range lists {
+			list, ok := l.(*v1.EventList)
+			if !ok {
+				continue
```

Comment:
Should we log or return an error here?

Reply:
It's not an error here, we only care about a specific for of events we query on
logging that error can flood the logs and will make it hard to track the real errors (it can be log debug for skipping this object which doesn't meet the query critera)